client/shared/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -237,7 +237,9 @@ object Client {
     *
     * @param app the [[HttpApp]] to respond to requests to this client
     */
-  def fromHttpApp[F[_]](app: HttpApp[F])(implicit F: Async[F]): Client[F] = {
+  def fromHttpApp[F[_]: Concurrent](
+      app: HttpApp[F]
+  )(implicit F: MonadCancelThrow[F]): Client[F] = {
```

Comment:
`Concurrent` extends `MonadCancelThrow`, so we can just ask for `Concurrent`.

```suggestion
  def fromHttpApp[F[_]](
      app: HttpApp[F]
  )(implicit F: Concurrent[F]): Client[F] = {
```